modules/tensor_mechanics/src/constraints/NodalStickConstraint.C

```diff
@@ -0,0 +1,298 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+// MOOSE includes
+#include "NodalStickConstraint.h"
+#include "NodalConstraintUtils.h"
+#include "MooseMesh.h"
+#include "Assembly.h"
+#include "SystemBase.h"
+
+#include "libmesh/mesh_inserter_iterator.h"
+#include "libmesh/parallel.h"
+#include "libmesh/parallel_elem.h"
+#include "libmesh/parallel_node.h"
+
+// C++ includes
+#include <limits.h>
+
+registerMooseObject("MooseApp", NodalStickConstraint);
+
+template <>
+InputParameters
+validParams<NodalStickConstraint>()
+{
+  InputParameters params = validParams<NodalConstraint>();
+  params.addRequiredParam<BoundaryName>("boundary", "The master boundary");
+  params.addRequiredParam<BoundaryName>("slave", "The slave boundary");
+  params.addRequiredParam<Real>("penalty", "Stiffness of the spring.");
+  return params;
+}
+
+NodalStickConstraint::NodalStickConstraint(const InputParameters & parameters)
+  : NodalConstraint(parameters),
+    _master_node_set_id(getParam<BoundaryName>("boundary")),
+    _slave_node_set_id(getParam<BoundaryName>("slave")),
+    _penalty(getParam<Real>("penalty"))
+{
+  updateConstrainedNodes();
+}
+
+void
+NodalStickConstraint::meshChanged()
+{
+  updateConstrainedNodes();
+}
+
+void
+NodalStickConstraint::updateConstrainedNodes()
+{
+  _master_node_vector.clear();
+  _connected_nodes.clear();
+  _master_conn.clear();
+
+  std::vector<dof_id_type> slave_nodelist =
+      _mesh.getNodeList(_mesh.getBoundaryID(_slave_node_set_id));
+  std::vector<dof_id_type> master_nodelist =
+      _mesh.getNodeList(_mesh.getBoundaryID(_master_node_set_id));
+
+  std::vector<dof_id_type>::iterator in;
```

Comment:
Same stuff I said about using `auto` in the other class.

It would be nice to share this code between the two classes, actually. There's enough duplicated code here that it's probably worth making a common base class for these two classes to facilitate that.  That code you moved to that util function could be in that class too.